picard/util/__init__.py

```diff
@@ -891,3 +894,38 @@ def wildcards_to_regex_pattern(pattern):
         regex.append('\\[')
         regex.append(wildcards_to_regex_pattern(''.join(group[1:])))
     return ''.join(regex)
+
+
+def unique_numbered_title(default_title, existing_titles):
+    """Generate a new unique and numbered title
+       based on given default title and existing titles
+    """
+    escaped = re.escape(default_title)
+    regex = re.compile('^' + escaped + '(?: \\((\\d+)\\))?$')
+    count = 0
+    for title in existing_titles:
+        m = regex.match(title)
+        if m:
+            num = m.group(1)
+            if num is not None:
+                count = max(count, int(num))
+            else:
+                count += 1
+    return "{0} ({1})".format(default_title, count + 1)
```

Comment:
I think to be more flexible with localization we should have the format string translated, e.g.

    "{title} ({count})".format(title=default_title, count=count + 1)

That allows localization to have this formatted as it best fits for the language, including replacing the `(` and `)` with different characters.

Reply:
I agree, but that will also require the regex on Line 904 to somehow be translated.  Any suggestions how that might be handled?

Reply:
Not trivial, we could perhaps parse the translated format string but how to handle `(?: )` part?
If the fmt string is reverse '({count}) {title}' for example, the optional part is anything but `{title}', so split in two parts:
`({count}) ` and `{title}`, replace `{count}` with  `(\\d+)` and wrap with `(?:` + `)`.
It would also mean translate the format string inside the method.

I'll give it a try.

Reply:
> It would also mean translate the format string inside the method.

And the regex used on Line 922 in the `get_base_title_suffix()` method.

Reply:
https://github.com/rdswift/picard/pull/11

Reply:
I didn't do `get_base_title_suffix()` but it probably be done the same way